workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.ts

```diff
@@ -0,0 +1,491 @@
+        .execute();
+    });
+    await Promise.all(batchDeleteRequestPromises);
+
+    return { deleted: true };
+  }
+
+  public async deleteSubjectPermissions(
+    deleteSubjectPermissionsRequest: DeleteSubjectPermissionsRequest
+  ): Promise<DeleteSubjectPermissionsResponse> {
+    this._isInit();
+    const { subjectType, subjectId } = deleteSubjectPermissionsRequest;
+
+    const { identityPermissions } = await this.getIdentityPermissionsBySubject({
+      subjectType,
+      subjectId
+    });
+    await this.deleteIdentityPermissions({
+      identityPermissions
+    });
+    return { deleted: true };
+  }
+
+  public async assignUserToGroup(
+    assignUserToGroupRequest: AssignUserToGroupRequest
+  ): Promise<AssignUserToGroupResponse> {
+    this._isInit();
+    const groupKey = {
+      pk: this._composeIdentityKey(this._groupPrefix, 'GROUP', assignUserToGroupRequest.groupId),
+      sk: this._composeIdentityKey(this._groupPrefix, 'GROUP', assignUserToGroupRequest.groupId)
+    };
+    const userGroupKey = {
+      pk: this._composeIdentityKey(this._assignedGroupPrefix, 'USER', assignUserToGroupRequest.userId),
+      sk: this._composeIdentityKey(this._assignedGroupPrefix, 'GROUP', assignUserToGroupRequest.groupId)
+    };
+
+    //Check if group exists
+    const response: GetItemCommandOutput = (await this._awsService.helpers.ddb
+      .get(groupKey)
+      .execute()) as GetItemCommandOutput;
+    if (!response || !response.Item) {
+      throw new GroupNotFoundError('Group not found');
+    }
+    //Assign user to group entry
+    await this._awsService.helpers.ddb.update(userGroupKey).execute();
+    return { assigned: true };
+  }
+
+  public async removeUserFromGroup(
+    removeUserFromGroupRequest: RemoveUserFromGroupRequest
+  ): Promise<RemoveUserFromGroupResponse> {
+    this._isInit();
+    const key = {
+      pk: this._composeIdentityKey(this._assignedGroupPrefix, 'USER', removeUserFromGroupRequest.userId),
+      sk: this._composeIdentityKey(this._assignedGroupPrefix, 'GROUP', removeUserFromGroupRequest.groupId)
+    };
+    await this._awsService.helpers.ddb.delete(key).execute();
+
+    return { removed: true };
+  }
+
+  public async getIdentityPermissionsBySubject(
+    getIdentityPermissionsBySubjectRequest: GetIdentityPermissionsBySubjectRequest
+  ): Promise<GetIdentityPermissionsBySubjectResponse> {
+    this._isInit();
+    const key = {
+      name: 'pk',
+      value: this._composeKey(
+        this._identityPermissionsPrefix,
+        getIdentityPermissionsBySubjectRequest.subjectType,
+        getIdentityPermissionsBySubjectRequest.subjectId
+      )
+    };
+
+    const query = this._awsService.helpers.ddb.query({
+      key
+    });
+
+    if (getIdentityPermissionsBySubjectRequest.action) {
+      query.filter('#a = :a');
+      query.names({ '#a': 'action' });
+      query.values(marshall({ ':a': getIdentityPermissionsBySubjectRequest.action }));
+    }
+    //IN operator maxed out at 100
```

Comment:
do we need to limit the `for` loop to max 100 iterations then?

Reply:
I think throwing an error could be better here but what are the scenarios where a user is assigned to 99 groups

Reply:
added a throughput exceeded error for when over 100 identities are presented